pkg/asset/machines/gcp/machines.go

```diff
@@ -108,3 +110,20 @@ func ConfigMasters(machines []machineapi.Machine, clusterID string) {
 		}
 	}
 }
+
+func getNetworks(platform *gcp.Platform, clusterID, role string) (string, string) {
```

Comment:
this is too ifelsey, a more flatter conditional structure is more readable.

```go
network := fmt.Sprintf("%s-network", clusterID)
if platform.ClusterNetwork != "" {
	network = platform.ClusterNetwork
}

subnetwork :=  fmt.Sprintf("%s-%s-subnet", clusterID, role)
if role == "worker" && platform.ComputeSubnet != "" {
	subnetwork = platform.ComputeSubnet
} 
if role == "master" && platform.ControlPlaneSubnet != "" {
	subnetwork = platform.ControlPlaneSubnet
}
```
And moving the role conditional to switch also might be more readable...